src/frontend/js/components/SolutionData.js

```diff
@@ -1,48 +1,245 @@
 import React from 'react'
 import { graphql, createFragmentContainer } from 'react-relay'
+import { Button, Chip } from 'material-ui'
+import { withStyles } from 'material-ui/styles'
 import renderSolution from '../helpers/vexFlowSolutionRendering'
 import { constructThemeSolutionData } from '../helpers/constructThemeSolutionData'
+import DeletionModal from './DeletionModal'
+import updateSolutionMutation from './updateSolutionMutation'
+import deleteSolutionMutation from './deleteSolutionMutation'
+
+const styles = theme => ({
+  icon: {
+    width: 35,
+    height: 35,
+    marginLeft: 5,
+    fontSize: 30,
+    backgroundColor: '#BEEBEF',
+    '&:hover': {
+      backgroundColor: '#BEEBEF'
+    },
+    '&:focus': {
+      backgroundColor: '#BEEBEF'
+    }
+  },
+  image: {
+    height: 21,
+    width: 23,
+    marginBottom: 0,
+    paddingBottom: 0
+  },
+  whole: {
+    height: 21,
+    width: 26,
+    marginBottom: 0,
+    padding: 0,
+    paddingLeft: 4
+  },
+  button: {
+    marginLeft: 10
+  }
+})
 
 class SolutionData extends React.Component {
   constructor () {
     super()
     this.state = {
       cursor: {
         stave: 0,
-        voice: 0,
+        voice: '',
         position: 0
-      }
+      },
+      themeData: '',
+      solutionData: '',
+      saveButton: 'save',
+      openModal: false
     }
     this.eventListener = this.eventListener.bind(this)
+    this.handleModification = this.handleModification.bind(this)
+    this.modifyDuration = this.modifyDuration.bind(this)
+    this.openModal = this.openModal.bind(this)
+    this.closeModal = this.closeModal.bind(this)
+    this.deleteSolution = this.deleteSolution.bind(this)
+  }
+
+  saveSolution () {
+    updateSolutionMutation.commit(this.props, this.state.solutionData)
+    this.setState({saveButton: 'Saved'})
+    setTimeout(() => { this.setState({saveButton: 'Save'}) }, 2000)
```

Comment:
This timeout should be cancelled if the user clicks the save button while still in the `Saved` state. Otherwise it can happen that button instantly changes from `Saved` to `Save`